doc/qbk/2.0.quicklook.qbk

```diff
@@ -0,0 +1,371 @@
+```
+
+The function __parse_uri__ returns an object of type `__result__<__url_view__>`
+which is a container resembling a variant that holds either an error or an object.
+
+[note
+    In this example we immediately call `result::value` which throws an exception
+    on a parsing error. The functions `result::has_value` and `result::has_error`
+    could also be used to check if the string has been parsed without throwing
+    errors.
+]
+
+When there are no errors, it `result::value` returns an instance of __url_view__,
+which holds the parsed result.
+
+It is worth noting that __parse_uri__ allocates no memory and throws no exceptions.
+Like a __string_view__, a __url_view__ does not retain ownership of the underlying
+string buffer. Constructed URL views always contain a valid URL in its correctly
+serialized form.
+
+[h5 Accessing]
+
+Accessing the parts of the URL is easy:
+
+[table [[Code][Output]] [[
+[c++]
+```
+    std::cout <<
+        "scheme    : " << u.``[link url.ref.boost__urls__url_view.scheme `scheme`]``()            << '\n' <<
+        "authority : " << u.``[link url.ref.boost__urls__url_view.encoded_authority `encoded_authority`]``() << '\n' <<
+        "path      : " << u.``[link url.ref.boost__urls__url_view.encoded_path `encoded_path`]``()      << '\n' <<
+        "query     : " << u.``[link url.ref.boost__urls__url_view.encoded_query `encoded_query`]``()     << '\n' <<
+        "fragment  : " << u.``[link url.ref.boost__urls__url_view.encoded_fragment `encoded_fragment`]``()  << '\n';
+```
+][
+[teletype]
+```
+    scheme    : https
+    authority : user:pass@www.example.com:443
+    path      : /path/to/my%2dfile.txt
+    query     : id=42&name=John%20Doe
+    fragment  : page%20anchor
+```
+]]]
+
+Functions like
+[link url.ref.boost__urls__url_view.encoded_query `encoded_query`]
+return string views, simply referencing the relevant portion of the
+URL string without the need to allocate memory.
+
+[h5 Decoding]
+
+To get a decoded string, call the same function without the word `encoded`:
+
+[table [[Code][Output]] [[
+[c++]
+```
+    std::cout <<
+        "query    : " << u.``[link url.ref.boost__urls__url_view.query `query`]``()            << '\n' <<
+        "fragment : " << u.``[link url.ref.boost__urls__url_view.fragment `fragment`]``()         << '\n';
+```
+][
+[teletype]
+```
+    query    : id=42&name=John Doe
+    fragment : page anchor
+```
+]]]
+
+The decoded functions return a __const_string__, which is a library type
+that models a read-only string. __const_string__ retains ownership of the
+underlying buffer and uses a type-erased allocator. An extra built-in
+buffer is also provided to avoid allocations for any string smaller than
+this [link url.ref.boost__urls__const_string.builtin_capacity `builtin_capacity`].
+
+[note
+    __url_view__ does not provide decoded functions for compound elements, such as
```

Comment:
This is weird though. The only element for which delimiters might not be ambiguous is `fragment` because it's the last one and usually not compound. The `url::query` example shows that clearly. We lose symmetry because some parts are being removed for ambiguity and some are not.

However, if we remove all the ambiguous ones, we are only left with the `encoded_` versions because all other decoded versions are ambiguous, except `fragment`. This is a problem because we have two options: (i) the `encoded_` prefix becomes redundant because it's all encoded, or (ii) the decoded versions need to be dedicated to the views, in which case and we would need to reformulate a few things.

Reply:
I don't think this is a problem